pkg/provisioner/ironic/ironic.go

```diff
@@ -709,7 +709,7 @@ func (p *ironicProvisioner) InspectHardware(force bool) (result provisioner.Resu
 		result, err = operationFailed(status.Error)
 		return
 	}
-	if !status.Finished || nodes.ProvisionState(ironicNode.ProvisionState) == nodes.InspectWait {
+	if !status.Finished || (nodes.ProvisionState(ironicNode.ProvisionState) == nodes.Inspecting || nodes.ProvisionState(ironicNode.ProvisionState) == nodes.InspectWait) {
```

Comment:
I wonder if this change should be done separately since (in openshift world we may want it backported since it seems to fix a bug).

Reply:
I can open a separate PR for that. Isn't BMO in Openshift in sync with upstream?

Reply:
Fixed it in https://github.com/metal3-io/baremetal-operator/pull/822, will remove it from here.